streaming/base/shared/prefix.py

```diff
@@ -1,28 +1,36 @@
 # Copyright 2023 MosaicML Streaming authors
 # SPDX-License-Identifier: Apache-2.0
 
-"""Synchronization primitives that live in shared memory.
+"""Register or look up the prefix to use for all shared resources.
 
-For when using `threading` or `multiprocessing` from the python standard library won't do, because
-we are coordinating separately instantiated pytorch worker processes.
+The prefix is used by all workers using this StreamingDataset of this training job. This is used to
+prevent shared resources like shared memory and filelocks from colliding.
 """
 
 from multiprocessing import resource_tracker  # pyright: ignore
 from multiprocessing.shared_memory import SharedMemory
-from typing import List, Set, Tuple
+from time import sleep
+from typing import Iterator, List, Set, Tuple
 
 import numpy as np
 import torch
 from torch import distributed as dist
 
-from streaming.base.shared.memory import CreateSharedMemory
+from streaming.base.shared import SharedMemory
+from streaming.base.util import TICK
 from streaming.base.world import World
 
-# Time to wait, in seconds.
-TICK = 0.07
 
-# Time out to wait before raising exception
-TIMEOUT = 60
+def _each_prefix_int() -> Iterator[int]:
+    """Get each possible prefix int to check in order.
+
+    Returns:
+        Iterator[int]: Each prefix int.
+    """
+    prefix_int = 0
+    while True:
```

Comment:
previously this loop was `range(10**6)`, should we maintain that behavior? Or is it intended for this to be infinite?

Reply:
infinite

low single digits in practice